pkg/istio-agent/agent.go

```diff
@@ -321,6 +322,15 @@ func (a *Agent) initializeEnvoyAgent(_ context.Context) error {
 		}
 		a.envoyOpts.ConfigPath = out
 		a.envoyOpts.ConfigCleanup = true
+
+		bytes, err := proto.Marshal(node.Metadata.ToStruct())
+		if err != nil {
+			return err
+		}
+		metadata := base64.RawStdEncoding.EncodeToString(bytes)
+		// Set metadata exchange environment variables
+		_ = os.Setenv("ISTIO_PEER_METADATA_ID", node.ID)
```

Comment:
should we set this on the envoy process call instead of in the ambient environment? This feels a bit sketchy

Reply:
Sure, I'm checking whether this is feasible - will clean up once it works.